pep_sphinx_extensions/pep_processor/parsing/pep_banner_directive.py

```diff
@@ -125,3 +125,39 @@ class CanonicalTypingSpecBanner(PEPBanner):
     admonition_class = nodes.attention
 
     css_classes = ["canonical-typing-spec", "sticky-banner"]
+
+
+class AbandonedBanner(PEPBanner):
+    """Insert an admonition for abandoned PEPs."""
+
+    admonition_class = nodes.warning
+    admonition_pre_template = "{link_content}"
+    admonition_pre_text = "This PEP has been abandoned."
+    css_classes = ["sticky-banner", "deprecated", "abandoned"]
+
+
+class RejectedBanner(PEPBanner):
+    """Insert an admonition for rejected PEPs."""
+
+    admonition_class = nodes.warning
+    admonition_pre_template = "{link_content}"
+    admonition_pre_text = "This PEP has been rejected."
+    css_classes = ["sticky-banner", "deprecated", "rejected"]
+
+
+class SupersededBanner(PEPBanner):
+    """Insert an admonition for superseded PEPs."""
+
+    admonition_class = nodes.warning
+    admonition_pre_template = "{link_content}"
+    admonition_pre_text = "This PEP has been superseded."
+    css_classes = ["sticky-banner", "deprecated", "superseded"]
+
+
+class WithdrawnBanner(PEPBanner):
+    """Insert an admonition for withdrawn PEPs."""
+
+    admonition_class = nodes.warning
+    admonition_pre_template = "{link_content}"
+    admonition_pre_text = "This PEP has been withdrawn."
+    css_classes = ["sticky-banner", "deprecated", "withdrawn"]
```

Comment:
Instead of duplicating it three times, maybe move the common logic into a superclass?

```suggestion
class DeprecatedBanner(PEPBanner):
    """Generic admonition for deprecated PEPs."""
    admonition_class = nodes.warning
    admonition_pre_template = "{link_content}"
    admonition_pre_text = "This PEP has been deprecated."
    css_classes = ["sticky-banner", "deprecated"]


class RejectedBanner(DeprecatedBanner):
    """Insert an admonition for rejected PEPs."""

    admonition_pre_text = "This PEP has been rejected."
    css_classes = ["sticky-banner", "deprecated", "rejected"]


class SupersededBanner(PEPBanner):
    """Insert an admonition for superseded PEPs."""

    admonition_pre_text = "This PEP has been superseded."
    css_classes = ["sticky-banner", "deprecated", "superseded"]


class WithdrawnBanner(PEPBanner):
    """Insert an admonition for withdrawn PEPs."""

    admonition_pre_text = "This PEP has been withdrawn."
    css_classes = ["sticky-banner", "deprecated", "withdrawn"]
```

Reply:
I considered it, but thought the duplication saved compared to the overhead was borderline (it results in more lines of code). Anyway, I've added a superclass in afa418d030ec16dca5c9083b5fac9165e676d154 so we can compare.